program/mobilenets-armcl-opencl/armcl_graph_common.h

```diff
@@ -0,0 +1,122 @@
+#include <arm_compute/graph/backends/CL/CLDeviceBackend.h>
+#else
+#include <arm_compute/graph/Graph.h>
+#include <arm_compute/graph/Nodes.h>
+#include <arm_compute/runtime/CL/CLScheduler.h>
+#endif
+
+#ifndef DATATYPE
+#define DATATYPE DataType::F32
+#endif
+
+inline void printf_callback(const char *buffer, unsigned int len, size_t complete, void *user_data) {
+  printf("%.*s", len, buffer);
+}
+
+inline void set_kernel_path() {
+  const char* kernel_path = getenv("CK_ENV_LIB_ARMCL_CL_KERNELS");
+  if (kernel_path) {
+    printf("Kernel path: %s\n", kernel_path);
+    arm_compute::CLKernelLibrary::get().set_kernel_path(kernel_path);
+  }
+}
+
+inline void init_armcl(arm_compute::ICLTuner *cl_tuner = nullptr) {
+  cl_context_properties properties[] =
+  {
+    CL_PRINTF_CALLBACK_ARM, reinterpret_cast<cl_context_properties>(printf_callback),
+    CL_PRINTF_BUFFERSIZE_ARM, static_cast<cl_context_properties>(0x100000),
+    CL_CONTEXT_PLATFORM, reinterpret_cast<cl_context_properties>(cl::Platform::get()()),
+    0
+  };
+  cl::Context::setDefault(cl::Context(CL_DEVICE_TYPE_DEFAULT, properties));
+  arm_compute::CLScheduler::get().default_init(cl_tuner);
+
+  // Should be called after initialization
+  set_kernel_path();
+
+#if defined(ARMCL_18_05_PLUS)
+  arm_compute::graph::backends::BackendRegistry::get()
+    .add_backend<arm_compute::graph::backends::CLDeviceBackend>(
+      arm_compute::graph::Target::CL);
+#endif
+}
+
+#if defined(ARMCL_18_05_PLUS)
+
+inline arm_compute::graph::DepthwiseConvolutionMethod str_to_dwsc_convolution_method(const char *method_name) {
+  // Try to get convolution method by its name
+  if (strcmp(method_name, "DEFAULT") == 0) return arm_compute::graph::DepthwiseConvolutionMethod::DEFAULT;
+  if (strcmp(method_name, "GEMV") == 0) return arm_compute::graph::DepthwiseConvolutionMethod::GEMV;
+  if (strcmp(method_name, "OPTIMIZED_3x3") == 0) return arm_compute::graph::DepthwiseConvolutionMethod::OPTIMIZED_3x3;
+  // Try to get convolution method as integer value.
+  return static_cast<arm_compute::graph::DepthwiseConvolutionMethod>(atoi(method_name));
+}
+
+inline arm_compute::graph::ConvolutionMethod str_to_convolution_method(const char *method_name) {
+  // Try to get convolution method by its name
+  if (strcmp(method_name, "DEFAULT") == 0) return arm_compute::graph::ConvolutionMethod::DEFAULT;
+  if (strcmp(method_name, "GEMM") == 0) return arm_compute::graph::ConvolutionMethod::GEMM;
+  if (strcmp(method_name, "DIRECT") == 0) return arm_compute::graph::ConvolutionMethod::DIRECT;
+  if (strcmp(method_name, "WINOGRAD") == 0) return arm_compute::graph::ConvolutionMethod::WINOGRAD;
+  // Try to get convolution method as integer value.
+  return static_cast<arm_compute::graph::ConvolutionMethod>(atoi(method_name));
+}
+
+inline arm_compute::graph::ConvolutionMethod get_convolution_method() {
+  auto method_name = getenv("CK_CONVOLUTION_METHOD");
+  if (!method_name) {
```

Comment:
A bit hard to read. How about:
```
if (method_name) {
    return str_to_convolution_method(method_name);
} else {
    if (arm_compute::CLScheduler::get().target() == arm_compute::GPUTarget::BIFROST) {
        return arm_compute::graph::ConvolutionMethod::DIRECT;
    } else {
        return arm_compute::graph::ConvolutionMethod::GEMM;
    }
}
```

Reply:
Fixed. But I personally don't like `else` when `if` returns. So if you don't insist... :)